nixos/modules/services/web-servers/apache-httpd/vhost-options.nix

```
@@ -47,10 +47,27 @@ in
       ];
       description = ''
         Listen addresses and ports for this virtual host.
-        <note><para>
+        <note>
+        <para>
           This option overrides <literal>addSSL</literal>, <literal>forceSSL</literal> and <literal>onlySSL</literal>.
-        </para></note>
+        </para>
+        <para>
+          If you only want to set the addresses manually and not the ports, take a look at <literal>listenAddresses</literal>.
+        </para>
+        </note>
+      '';
+    };
+
+    listenAddresses = mkOption {
+      type = with types; listOf str;
```

```suggestion
      type = with types; nonEmptyListOf str;
```

In conjunction with the changed default below.

---

Awesome! I think I recall seeing `nonEmptyListOf` being introduced, but completely forgot about it. Much appreciated.